runtime/near-vm-runner/src/runner.rs

```diff
@@ -112,86 +44,80 @@ pub fn run_vm(
             promise_results,
             current_protocol_version,
             cache,
-        ),
-        #[cfg(not(feature = "wasmer2_vm"))]
-        VMKind::Wasmer2 => panic!("Wasmer2 is not supported, compile with '--features wasmer2_vm'"),
-    };
-    (outcome, error)
+        )
+    } else {
+        panic!("the {:?} runtime has not been enabled at compile time", vm_kind);
+    }
 }
 
-/// `precompile` compiles WASM contract to a VM specific format and stores result into the `cache`.
-/// Further execution with the same cache will result in compilation avoidance and reusing cached
-/// result. `wasm_config` is required as during compilation we decide if gas metering shall be
-/// embedded in the native code, and so we take that into account when computing database key.
-#[allow(dead_code)]
-pub fn precompile<'a>(
-    code: &[u8],
-    code_hash: &CryptoHash,
-    wasm_config: &'a VMConfig,
-    cache: &'a dyn CompiledContractCache,
-    vm_kind: VMKind,
-) -> Option<VMError> {
-    match vm_kind {
-        #[cfg(not(feature = "wasmer0_vm"))]
-        VMKind::Wasmer0 => panic!("Wasmer0 is not supported, compile with '--features wasmer0_vm'"),
-        #[cfg(feature = "wasmer0_vm")]
-        VMKind::Wasmer0 => {
-            let result = crate::cache::wasmer0_cache::compile_and_serialize_wasmer(
-                code,
-                wasm_config,
-                code_hash,
-                cache,
-            );
-            into_vm_result(result).err()
-        }
-        #[cfg(feature = "wasmer2_vm")]
-        VMKind::Wasmer2 => {
-            let store = crate::wasmer2_runner::default_wasmer2_store();
-            let result = crate::cache::wasmer2_cache::compile_and_serialize_wasmer2(
-                code,
-                code_hash,
-                wasm_config,
-                cache,
-                &store,
-            );
-            into_vm_result(result).err()
-        }
-        #[cfg(not(feature = "wasmer2_vm"))]
-        VMKind::Wasmer2 => panic!("Wasmer2 is not supported, compile with '--features wasmer2_vm'"),
-        VMKind::Wasmtime => Some(VMError::FunctionCallError(FunctionCallError::CompilationError(
-            CompilationError::UnsupportedCompiler {
-                msg: "Precompilation not supported in Wasmtime yet".to_string(),
-            },
-        ))),
-    }
+pub trait Runtime {
```

Comment:
It's relatively important to pick a different name here, as we already have one `Runtime` in nearcore, and I fear having two would lead to confusion. Two of the possible options:  `Vm` or `ContractRuntime`.